ynput/OpenPype
# Blender: Set --python-use-system-env from PreLaunchHook

## Changelog Description

Automatically add `--python-use-system-env` when running Blender when it's not pre-configured in the application settings for Blender (In Studio Settings > System Settings > Applications > Blender > {version} > arguments.

## Additional info

It often happens that admins add a new Blender version in the applications but don't also apply the `--python-use-system-env`  arguments meaning OpenPype will not be loaded for Blender plus won't be available in Blender on `PYTHONPATH`.

This avoids that issue by always enforcing the argument is added on launch plus updates the defaults so that `--python-use-system-env` isn't there by default either.

@iLLiCiTiT I've set it to run on `launch_types = set()` so it always runs. I basically want it to run on ANY Blender launch where prior to this it'd have also been applied due to being in the arguments settings for the application. (I feel like that's likely more than just 'local', right?)

---

Before:
![image](https://github.com/ynput/OpenPype/assets/2439881/df39db46-a81b-4f50-b537-9140ac379f37)


After:
![image](https://github.com/ynput/OpenPype/assets/2439881/70ba84e1-3276-486e-babc-39f6c3626f22)

---

See [this reference discussion on Discord for somebody who was hitting this issue](https://discord.com/channels/517362899170230292/517382145552154634/1148250026099875990). Although there have also been [reports that it should be working without the flag](https://discord.com/channels/517362899170230292/563751989075378201/943193907842404362).

**Note:** For those who were already succsesfully launching blender without `--python-use-system-env` this PR might be giving unexpected results if - for whatever reason - they don't actually use the system env in Blender.


## Testing notes:

1. Run blender without `--python-use-system-env` argument in settings
2. Run blender with `--python-use-system-env` argument in settings (this is just to ensure it's backwards compatible with any settings a studio might currently have for custom versions.)
3. Everything should load and run fine for both
